main.tf

```diff
@@ -26,7 +26,7 @@ resource "aws_security_group_rule" "egress" {
 }
 
 resource "aws_security_group_rule" "http_ingress" {
-  count             = "${var.http_enabled}"
+  count             = "${var.http_enabled ? 1 : 0}"
```

Comment:
use explicit comparison: `var.http_enabled == "true"`

Reply:
make sure `http_enabled` is a string. boolean variables in TF are not recommended by terraform.